### PR TITLE
Update 'Rename's commit behavior to be async.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dibarbet @ryzngard for review please :)

File: src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs [hunk 1]
```diff
@@ -827,23 +875,25 @@ private void CommitCore(IUIThreadOperationContext operationContext, bool preview
...
                 {
-                    // We pass CancellationToken.None here because we don't have a usable token to pass. The IUIThreadOperationContext
-                    // passed here as a cancellation token, but the caller in CommitCore has already turned off cancellation
-                    // because we're committed to the update at this point. If we ever want to pass cancellation here, we'd want to move this
-                    // part back out of this method and before the point where we've already opened a global transaction.
-                    var root = newDocument.GetSyntaxRootSynchronously(CancellationToken.None);
+                    // We pass CancellationToken.None here because we don't have a usable token to pass. The
+                    // IUIThreadOperationContext passed here as a cancellation token, but the caller in CommitCore has
+                    // already turned off cancellation because we're committed to the update at this point. If we ever
+                    // want to pass cancellation here, we'd want to move this part back out of this method and before
```
Comment (dibarbet) [inline src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs:881]: comment seems off - says we pass None, but we just pass in the callers token

**Review by dibarbet — APPROVED** [review]

File: src/EditorFeatures/Core/IInlineRenameSession.cs [hunk 2]
```diff
@@ -47,6 +51,10 @@ internal interface IInlineRenameSession
         /// <summary>
         /// Dismisses the rename session, completing the rename operation across all files.
         /// </summary>
+        [Obsolete($"Use {nameof(CommitAsync)} instead", error: true)]
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/IInlineRenameSession.cs:54]: nope.  i'll just remove :)

File: src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs]: fixed.

**Pull request merged**